code/extensions/Lumberjack.php

```diff
@@ -16,14 +16,15 @@ class Lumberjack extends SiteTreeExtension
     /**
      * Loops through subclasses of the owner (intended to be SiteTree) and checks if they've been hidden.
      *
+     * @var bool $allowSiteTreeVisibility       whether to allow both gridfield and site tree visibility
      * @return array
      **/
-    public function getExcludedSiteTreeClassNames()
+    public function getExcludedSiteTreeClassNames($allowSiteTreeVisibility = false)
     {
         $classes = array();
         $siteTreeClasses = $this->owner->allowedChildren();
         foreach ($siteTreeClasses as $class) {
-            if (Config::inst()->get($class, 'show_in_sitetree') === false) {
+            if ((Config::inst()->get($class, 'show_in_sitetree') === false) || ($allowSiteTreeVisibility && Config::inst()->get($class, 'show_in_lumberjack'))) {
```

Comment:
I don't think you need to check the config value twice, especially as in one check you're looking for type matching with false and the other you are loosely checking for true. TBH, I'm not really sure what this behaviour is meant to be, if show in sitetree is false it's pushing it or if it's true AND `$allowSiteTreeVisibility` is true it'll push it... which seems wrong?

Can't we do `if (Config::inst()->get($class, 'show_in_sitetree') && $allowSiteTreeVisibility)`?